docs/examples/ipfs-101/README.md

```diff
@@ -0,0 +1,5 @@
+# IPFS 101, spawn a node and add a file to the IPFS network
+
+> This example is the sister example of IPFS 101 for [js-ipfs](https://github.com/ipfs/js-ipfs/tree/master/examples/ipfs-101).
+
+In this tutorial, we go through spawning an IPFS node, adding a file and cat'ing the file multihash locally and through the gateway.
```

Comment:
I'd add a _big_ caveat about how all these interfaces are internal and unstable. The constructor is _not_ done.

Also, we should probably call this something like "how to use IPFS as a library". When people think about "how to spawn an IPFS node", they're usually wondering how to spawn the daemon.

Reply:
> I'd add a big caveat about how all these interfaces are internal and unstable. The constructor is not done.

Even the ones that are Core API?

> Also, we should probably call this something like "how to use IPFS as a library". When people think about "how to spawn an IPFS node", they're usually wondering how to spawn the daemon.

I was really looking to have a reliable way to use IPFS as a library by being able to spawn a in-process node and not spinning an extra process as a daemon.

Are you highlighting that is not a timely cause at the moment? If yes, why isn't it or why wouldn't it be?

Reply:
* The _core_ API is pretty stable but we still have a few things we'd like to change.
* The _constructor_ is not stable. Magik6k was working on that before he got pulled into a different project.

> Are you highlighting that is not a timely cause at the moment? If yes, why isn't it or why wouldn't it be?

I'm saying that this is really confusing. _Most_ users are trying to use go-ipfs ad a daemon. If we introduce a "how to use IPFS 101" tutorial that tells them how to use it in a process, they're going to be _very_ confused.

And yes, until we ship the new constructor (https://github.com/libp2p/specs/issues/215), I'd like to be careful about telling users to do this (at least warn them).

Reply:
> If we introduce a "how to use IPFS 101" tutorial that tells them how to use it in a process, they're going to be very confused.

Oh, got, this is an new perspective for me. My intuition tells me that it is more obvious that an `import package` gives me a way to spawn things in-process rather than creating a separate process, and I guess the reason being that anything that tries to manage a process for you is either bloated with complexity or too naive to last. 

There is one package (super useful) in js that does let a user spawn any kind of daemon (both go and js) https://github.com/ipfs/js-ipfsd-ctl/ and it has proven to be super convenient. We can potentially do both, I would be super fine with that.

Reply:
We're still not on the same page. Go-ipfs is currently an _application_, first and foremost. My objection is  that "ipfs 101" is synonymous with "introduction to using go-ipfs". That implies that this is the expected/primary way to use go-ipfs.

I'm just saying that we need to change this to: "How to use go-ipfs as a library"

Reply:
Ah! Ok, makes sense then :)

Reply:
Updated :)